File: eio2024-lv/2024-10-07-lv/buss/solution/sol.py
# loeme sisendist praeguse kellaaja ning teisendame selle minutiteks
H, M = [int(x) for x in input().split()]
C = H * 60 + M

peatus = -1 # muutuja hetkel suurima ajavaruga peatuse meeles pidamiseks
ajavaru = -10001 # muutuja maksimaalse ajavaru jaoks

N = int(input())
for i in range(1, N + 1):
	H, M, J = [int(x) for x in input().split()]
	B = 60 * H + M # teisendame bussi saabumisaja minutiteks
	varu = (B - C) - J # arvutame, mitu minutit varem jõuab Juku
	if varu > ajavaru: # kui leidsime parema ajavaruga peatuse, siis uuendame praeguseid parimad
		ajavaru = varu
		peatus = i

# väljund
if ajavaru >= 0:
	print("JAH")
else:
	print("EI")
print(peatus)

File: eio2024-lv/2024-10-07-lv/buss/solution/sol.cpp
#include <iostream>
using namespace std;

int main() {
	// loeme sisendist praeguse kellaaja ning teisendame selle minutiteks
	int H, M;
	cin >> H >> M;
	const int C = H * 60 + M;

	int ajavaru = -10001; // muutuja maksimaalse ajavaru jaoks
	int peatus = -1; // muutuja seni suurima ajavaruga peatuse meeles pidamiseks

	int N;
	cin >> N;
	for (int i = 1; i <= N; ++i) {
		int J;
		cin >> H >> M >> J;
		const int B = H * 60 + M; // teisendame bussi saabumisaja minutiteks
		const int varu = (B - C) - J; // arvutame, mitu minutit varem jõuab Juku
		if (varu > ajavaru) { // kui leidsime parema ajavaru, siis uuendame
			ajavaru = varu;
			peatus = i;
		}
	}
	// väljund
	if (ajavaru >= 0) {
		cout << "JAH\n";
	} else {
		cout << "EI\n";
	}
	cout << peatus << '\n';
}
